src/AWS4AuthRequest.jl

```diff
@@ -90,10 +101,14 @@ function sign_aws4!(method::String,
     end
 
     # Sort and lowercase() Headers to produce canonical form...
-    canonical_headers = sort!(map(headers) do (k, v)
-        string(lowercase(k), ':', replace(strip(v), r"\s+" => " "))
-    end)
-    signed_headers = join(sort!(map(lowercase∘first, headers)), ";")
+    unique_headers = sort!(unique!(map(lowercase∘first, headers)))
+    canonical_headers = map(unique_headers) do k
+        # All values present for this key (case-insensitive) with whitespace normalized
+        vs = map(i->_normalize_ws(last(headers[i])),
+                 findall(x->lowercase(first(x)) == k, headers))
```

Comment:
This map + findall is kind scary
I think just writing out the for loop is cleaner

Something like: 
```
vs = Vector{Header}() # is this eltype right?
for header in headers
	header_key = lowercase(first(header))
	if header_key = k
		header_value
		push!(vs, _normalize_ws(header_value))
	end
end
```